src/machines/java/com/enderio/machines/client/rendering/blockentity/CapacitorBankBER.java

```diff
@@ -0,0 +1,316 @@
+import net.minecraft.client.gui.Font;
+import net.minecraft.client.renderer.LevelRenderer;
+import net.minecraft.client.renderer.LightTexture;
+import net.minecraft.client.renderer.MultiBufferSource;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.client.renderer.blockentity.BlockEntityRenderer;
+import net.minecraft.client.renderer.blockentity.BlockEntityRendererProvider;
+import net.minecraft.client.renderer.texture.TextureAtlasSprite;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.resources.ResourceLocation;
+import net.minecraft.world.inventory.InventoryMenu;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.Block;
+import org.joml.Matrix3f;
+import org.joml.Matrix4f;
+import org.joml.Vector2i;
+
+public class CapacitorBankBER implements BlockEntityRenderer<CapacitorBankBlockEntity> {
+
+    private static final ResourceLocation FULL_BAR = EnderIO.loc("block/capacitor_additionals/capacitor_bank_bar_full");
+    private static final ResourceLocation END_BAR = EnderIO.loc("block/capacitor_additionals/capacitor_bank_bar_end");
+    private static final ResourceLocation ENERGY_BAR = EnderIO.loc("block/capacitor_additionals/capacitor_bank_bar_energy");
+
+    private static final ResourceLocation IO_1x1 = EnderIO.loc("block/capacitor_additionals/1x1_full");
+    private static final ResourceLocation IO_FULL = EnderIO.loc("block/capacitor_additionals/full");
+    public CapacitorBankBER(BlockEntityRendererProvider.Context context) {
+
+    }
+
+    @Override
+    public void render(CapacitorBankBlockEntity blockEntity, float partialTick, PoseStack poseStack, MultiBufferSource bufferSource, int packedLight,
+        int packedOverlay) {
+
+        if (blockEntity.getLevel() != null) {
+            for (Direction direction: new Direction[] {Direction.NORTH, Direction.EAST, Direction.SOUTH, Direction.WEST}) {
+                BlockPos facingPos = blockEntity.getBlockPos().relative(direction);
+                if (Block.shouldRenderFace(blockEntity.getBlockState(), blockEntity.getLevel(), blockEntity.getBlockPos(), direction, facingPos)) {
+                    DisplayMode mode = blockEntity.getDisplayMode(direction);
+                    if (mode == DisplayMode.BAR) {
+                        renderBar(blockEntity, poseStack, bufferSource, direction);
+                    }
+                    if (mode == DisplayMode.IO) {
+                        renderIO(blockEntity, poseStack, bufferSource, direction);
+                    }
+                }
+            }
+        }
+    }
+
+    private static void renderBar(CapacitorBankBlockEntity blockEntity, PoseStack poseStack, MultiBufferSource bufferSource, Direction facing) {
+        if (getDisplayModeRelative(blockEntity, facing, new Vector2i(0,-1)) == DisplayMode.BAR) {
+            return;
+        }
+        int length = 1;
+        //some random limit
+        while(length < 16) {
```

Comment:
What is this limit?

Reply:
I think how many screens/bars can connect to each other?

Reply:
yeah, basically, I don't want the BER to be as computationally expensive and as the energy bar is rendered from the topmost one, I would have to make the BlockEntity Render Box larger and larger which would just cost more on the rendering side. I thought that 16 blocks is a reasonable limit